src/common/mfu_daos.c

```diff
@@ -1313,3 +1330,2195 @@ int mfu_flist_copy_daos(
+                rc = 1;
+                goto out;
+            }
+
+            /* TODO: remove random checking/printing to make sure
+             * right number of blocks is selected
+             */
+            hsize_t sel = H5Sget_select_npoints(hdf5->rx_dspace);
+            printf("SEL: %d\n", (int)sel);
+            hsize_t mem_sel = H5Sget_select_npoints(hdf5->rx_memspace);
+            printf("MEM SEL: %d\n", (int)mem_sel);
+            hssize_t nblocks = H5Sget_select_hyper_nblocks(hdf5->rx_dspace);
+            printf("NUM BLOCKS SELECTED: %d\n", (int)nblocks);
+            htri_t valid = H5Sselect_valid(hdf5->rx_dspace);
+            printf("VALID: %d\n", (int)valid);
+
+            hdf5->rx_dtype = H5Tcreate(H5T_OPAQUE, (*iod).iod_size);
+            if (hdf5->rx_dtype < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to create rx_dtype");
+                rc = 1;
+                goto out;
+            }
+            /* HDF5 should not try to interpret the datatype */
+            status = H5Tset_tag(hdf5->rx_dtype, "Opaque dtype");
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to set dtype tag");
+                rc = 1;
+                goto out;
+            }
+            status = H5Dwrite(hdf5->rx_dset, hdf5->rx_dtype,
+                              hdf5->rx_memspace, hdf5->rx_dspace,
+                              H5P_DEFAULT, sgl.sg_iovs[0].iov_buf);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to write rx_dset");
+                rc = 1;
+                goto out;
+            }
+            printf("\tRECX DSET ID WRITTEN: %d\n",
+                (int)hdf5->rx_dset);
+            printf("\tRECX DSPACE ID WRITTEN: %d\n",
+                (int)hdf5->rx_dspace);
+            /* get size of buffer needed
+             * from nalloc
+             */
+            status = H5Sencode(hdf5->rx_dspace, NULL, &nalloc);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to get size of buffer needed");
+                rc = 1;
+                goto out;
+            }
+            /* encode dataspace description
+             * in buffer then store in
+             * attribute on dataset
+             */
+            encode_buf = malloc(nalloc * sizeof(unsigned char));
+            status = H5Sencode(hdf5->rx_dspace, encode_buf,
+                               &nalloc);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to encode dataspace");
+                rc = 1;
+                goto out;
+            }
+            /* created attribute in HDF5 file with encoded
+             * dataspace for this record extent */
+            memset(attr_name, 64, sizeof(attr_name));
+            memset(number_str, 16, sizeof(number_str));
+            memset(attr_num_str, 16, sizeof(attr_num_str));
+            path_len = snprintf(number_str, 10, "%d",
+                        (int)(*ak_index));
+            if (path_len >= 16) {
```

Comment:
@dsikich Why 16? Also, for these 16 and 64 uses, maybe we should have a #define in case we need to change the value?